src/client/iof/src/common/iof_rpc.c

```diff
@@ -0,0 +1,484 @@
+	struct ios_gah *data = arg;
+
+	return crt_proc_memcpy(proc, data, sizeof(*data));
+}
+
+int
+iof_proc_stat(crt_proc_t proc, void *arg)
+{
+	struct stat *data = arg;
+
+	return crt_proc_memcpy(proc, data, sizeof(*data));
+}
+
+struct crt_msg_field CMF_IOF_NAME = {
+	.cmf_size = sizeof(struct ios_name),
+	.cmf_proc = crt_proc_struct_ios_name,
+};
+
+struct crt_msg_field CMF_IOF_STAT = {
+	.cmf_size = sizeof(struct stat),
+	.cmf_proc = iof_proc_stat,
+};
+
+struct crt_msg_field *gah_string_in[] = {
+	&CMF_GAH,	/* gah */
+	&CMF_IOF_NAME,	/* name */
+};
+
+struct crt_msg_field *imigrate_in[] = {
+	&CMF_GAH,	/* gah of parent */
+	&CMF_IOF_NAME,	/* name */
+	&CMF_INT,	/* inode */
+};
+
+struct crt_msg_field *string_out[] = {
+	&CMF_STRING,
+	&CMF_INT,
+	&CMF_INT,
+};
+
+struct crt_msg_field *entry_out[] = {
+	&CMF_GAH,	/* gah */
+	&CMF_IOF_STAT,	/* struct stat */
+	&CMF_INT,	/* rc */
+	&CMF_INT,	/* err */
+};
+
+struct crt_msg_field *create_out[] = {
+	&CMF_GAH,	/* gah */
+	&CMF_GAH,	/* inode gah */
+	&CMF_IOF_STAT,	/* struct stat */
+	&CMF_INT,	/* rc */
+	&CMF_INT,	/* err */
+};
+
+struct crt_msg_field *two_string_in[] = {
+	&CMF_GAH,
+	&CMF_IOF_NAME,
+	&CMF_STRING,
+};
+
+struct crt_msg_field *create_in[] = {
+	&CMF_GAH,	/* gah */
+	&CMF_IOF_NAME,	/* name */
+	&CMF_INT,	/* mode */
+	&CMF_INT,	/* flags */
+};
+
+struct crt_msg_field *rename_in[] = {
+	&CMF_GAH,	/* old parent */
+	&CMF_GAH,	/* new parent */
+	&CMF_IOF_NAME,	/* old name */
+	&CMF_IOF_NAME,	/* new name */
+	&CMF_INT,	/* flags */
+};
+
+struct crt_msg_field *open_in[] = {
+	&CMF_GAH,	/* gah */
+	&CMF_INT,	/* flags */
+};
+
+struct crt_msg_field *unlink_in[] = {
+	&CMF_IOF_NAME,	/* name */
+	&CMF_GAH,	/* gah */
+	&CMF_INT,	/* flags */
+};
+
+struct crt_msg_field *attr_out[] = {
+	&CMF_IOF_STAT,	/* stat */
+	&CMF_INT,	/* rc */
+	&CMF_INT,	/* err */
+};
+
+struct crt_msg_field *iov_pair[] = {
+	&CMF_IOVEC,
+	&CMF_INT,
+	&CMF_INT
+};
+
+struct crt_msg_field *gah_pair[] = {
+	&CMF_GAH,
+	&CMF_INT,
+	&CMF_INT
+};
+
+struct crt_msg_field *readdir_in[] = {
+	&CMF_GAH,
+	&CMF_BULK,
+	&CMF_UINT64,
+};
+
+struct crt_msg_field *readdir_out[] = {
+	&CMF_IOVEC,
+	&CMF_INT,
+	&CMF_INT,
+	&CMF_INT,
+	&CMF_INT,
+};
+
+CRT_GEN_PROC_FUNC(iof_xtvec, IOF_STRUCT_XTVEC);
+
+CRT_RPC_DEFINE(iof_readx, IOF_RPC_READX_IN, IOF_RPC_READX_OUT);
+CRT_RPC_DEFINE(iof_writex, IOF_RPC_WRITEX_IN, IOF_RPC_WRITEX_OUT);
+
+struct crt_msg_field *status_out[] = {
+	&CMF_INT,
+	&CMF_INT,
+};
+
+struct crt_msg_field *gah_in[] = {
+	&CMF_GAH,
+};
+
+struct crt_msg_field *writex_in[] = {
+	&CMF_GAH,
+	&CMF_IOVEC,
+	&CMF_UINT64,
+	&CMF_UINT64,
+	&CMF_UINT64,
+	&CMF_UINT64,
+	&CMF_BULK,
+	&CMF_BULK,
+};
+
+struct crt_msg_field *writex_out[] = {
+	&CMF_UINT64,
+	&CMF_INT,
+	&CMF_INT,
+	&CMF_UINT64,
+	&CMF_UINT64,
+};
+
+struct crt_msg_field *setattr_in[] = {
+	&CMF_GAH,	/* gah */
+	&CMF_IOF_STAT,	/* struct stat */
+	&CMF_UINT32,	/* to_set */
+};
+
+CRT_GEN_PROC_FUNC(iof_fs_info, IOF_FS_INFO)
+
+CRT_RPC_DEFINE(iof_query, ,IOF_SQ_OUT)
```

Comment:
(style)  space prohibited before that ',' (ctx:WxV)

(style)  space required after that ',' (ctx:WxV)